CoreArrangementPlugins/test/unit/src/au/gov/asd/tac/constellation/plugins/arrangements/hierarchical/HierarchicalArrangerNGTest.java

```
@@ -0,0 +1,319 @@
+            List<Integer> childLevel = new ArrayList<>();
+            for (int i = 0; i < 4; i++) {
+                parentLevel.add(vtxList.get(vtxIndex++));
+            }
+            while (vtxIndex < vtxList.size()) {
+                for (Integer parentId : parentLevel) {
+                    childLevel.add(vtxList.get(vtxIndex++));
+                    if (vtxIndex >= vtxList.size()) {
+                        break;
+                    }
+                    childLevel.add(vtxList.get(vtxIndex++));
+                    if (vtxIndex >= vtxList.size()) {
+                        break;
+                    }
+                }
+                int childPos = 0;
+                for (Integer parentId : parentLevel) {
+                    // link parents to children
+                    txnList.add(wg.addTransaction(parentId, childLevel.get(childPos++), true));
+                    if (childPos >= childLevel.size()) {
+                        childPos = 0;
+                    }
+                    txnList.add(wg.addTransaction(parentId, childLevel.get(childPos++), true));
+                    childPos++;
+                    if (childPos >= childLevel.size()) {
+                        childPos = 0;
+                    }
+                }    
+                parentLevel = childLevel;                
+                childLevel = new ArrayList<>();
+                // remove some entries in the parentLevel ... not all the children will be parents ... you know, diversity and all
+                int parentCount = parentLevel.size();
+                for (int c = parentCount - 1; c > -1; c--) {
+                    if (c%3 == 0) {
+                        parentLevel.remove(c);
+                    }
+                }
+            }
+            
+            wg.commit();
+        } catch (final InterruptedException ex) {
+            Exceptions.printStackTrace(ex);
+            Thread.currentThread().interrupt();
+        }
```

few instances like this one: given nothing happens after this, might be better to throw the exception here, rather than catch